src/main/scala/sbtghactions/GenerativePlugin.scala

```diff
@@ -783,13 +783,72 @@ ${indent(jobs.map(compileJob(_, sbt)).mkString("\n\n"), 1)}
       sbt)
   }
 
-  private val readCleanContents = Def task {
-    val src = Source.fromURL(getClass.getResource("/clean.yml"))
-    try {
-      src.mkString
-    } finally {
-      src.close()
-    }
+  private def generateCleanContents(runsOnOs: String) = {
```

Comment:
It can be argued that having one single giant string is easier to work with rather than splitting it up into multiple variables as its done  however since `$` is a reserved character for string interpolation and its very commonly used in github workflow `.yml` files all of the escaping would be tedious and somewhat confusing so I opted to do it this way.